src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerAnalyzer.cs

```diff
@@ -198,48 +204,4 @@ protected sealed override bool ShouldAnalyze(CancellationToken cancellationToken
 
         return addMethods.Any(static m => m is IMethodSymbol methodSymbol && methodSymbol.Parameters.Any());
     }
-
-    private bool TryAnalyzeIndexAssignment(
```

Comment:
moved to helper in a place where CE code could use it.  previously we only supported converting index-assignmetns to a collection-INITIALIZER.  now that collection-EXPRESSIONs support this, we want to use the same helper from both places.